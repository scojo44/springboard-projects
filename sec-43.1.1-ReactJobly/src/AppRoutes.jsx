import React from 'react'
import {Routes, Route} from 'react-router-dom'
import UserRoutes from './UserRoutes'
import Home from './Home'
import CompanyList from './company/CompanyList'
import CompanyDetail from './company/CompanyDetail'
import JobList from './job/JobList'
import LoginForm from './user/LoginForm'
import SignupForm from './user/SignupForm'
import ProfileForm from './user/ProfileForm'

export default function AppRoutes({login, signup, updateUser}) {
  return (
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="/login" element={<LoginForm login={login} />} />
      <Route path="/signup" element={<SignupForm signup={signup} />} />
      <Route element={<UserRoutes />}>
        <Route path="/companies" element={<CompanyList />} />
        <Route path="/companies/:handle" element={<CompanyDetail />} />
        <Route path="/jobs" element={<JobList />} />
        <Route path="/profile" element={<ProfileForm update={updateUser} />} />
      </Route>
    </Routes>
  );
}
