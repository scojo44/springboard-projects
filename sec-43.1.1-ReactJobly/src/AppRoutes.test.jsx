import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import UserContext from './UserContext'
import AppRoutes from './AppRoutes'

const context = {
  appliedJobIDs: new Set([1]),
  applyToJob: vi.fn(),
  currentUser: {
    username: 'testuser',
    firstName: 'First',
    lastname: 'Last',
    email: 'first@last.test',
    jobs: []
  }
};

describe('AppRoutes Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <AppRoutes />
        </UserContext.Provider>
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <AppRoutes />
        </UserContext.Provider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
