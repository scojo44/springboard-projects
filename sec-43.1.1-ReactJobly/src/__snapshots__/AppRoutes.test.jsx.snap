// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppRoutes Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="Home"
  >
    <h2>
      Welcome, guest!
    </h2>
  </section>
</DocumentFragment>
`;
